python/tvm/relax/op/transform.py

```diff
@@ -338,33 +343,130 @@ def full(
     dtype : Optional[str]
         The data type of the target.
 
+        If dtype is not given, the dtype of the target would be the dtype of fill_value.
+
     Returns
     -------
     result : relax.Expr
-        The resulting tensor.
+        The result tensor.
     """
-    if isinstance(shape, (PrimExpr, int)):
-        shape = [shape]
-    if isinstance(shape, (tuple, list)):
-        temp_shape = []
-        for shape in shape:
-            if isinstance(shape, PrimExpr):
-                temp_shape.append(shape)
-            elif isinstance(shape, int):
-                temp_shape.append(tvm.tir.const(shape, "int64"))
-            else:
-                raise RuntimeError(
-                    f"The input new shape of reshape operator contains unrecognized dimension {shape}"
-                )
-        shape = relax.ShapeExpr(temp_shape)
+    shape = _handle_shape(shape)
 
     if dtype is None:
-        dtype = tvm.DataType("void")
+        # currently tvm.DataType cannot identify "void"
+        # so leave dtype to be None so it will be converted into void type in PackedFunc calling process
+        # dtype = tvm.DataType("void")
+        dtype = None
     elif isinstance(dtype, str):
         dtype = tvm.DataType(dtype)
     return _ffi_api.full(fill_value, shape, dtype)
 
 
+def full_like(data: Expr, fill_value: Expr) -> Expr:
+    """Return a scalar value array of the same shape and type as data, filled with fill_value.
+
+    Parameters
+    ----------
+    data : relax.Expr
+        The input tensor.
+
+    fill_value : relax.Expr
+        The scalar value to fill.
+
+    Returns
+    -------
+    result : relax.Expr
+        The result tensor.
+    """
+    return _ffi_api.full_like(data, fill_value)
+
+
+def ones(
+    shape: Union[PrimExprLike, List[PrimExprLike], Tuple[PrimExprLike], Expr],
+    dtype: Optional[Union[str, tvm.DataType]] = "float32"
+) -> Expr:
+    """Fill array of given shape and dtype with ones.
+
+    Parameters
+    ----------
+    shape : Union[PrimExprLike, List[PrimExprLike], Tuple[PrimExprLike], Expr]
+        The shape of the target.
+
+    dtype : Optional[str]
+        The data type of the target.
+
+        If dtype is not given, the dtype of the target would be float32.
```

Comment:
Ditto. No need for such a blank line.
```suggestion
        The data type of the target.
        If dtype is not given, the dtype of the target would be float32.
```